Installer/source/main.cpp

```diff
@@ -1704,9 +1704,17 @@ else
 		}
 		else
 		{
-			printf("\x1b[2J");
-			fflush(stdout);
-			abort_pre_init("\r\n\r\nfailed to do AHBPROT magic: error %d\r\n",ios_patched);
+			if ((*(volatile unsigned int *)(0xcd8005a0) >> 16 ) == 0xffff)

```

Comment:
this register address is afaik out of bound on a regular wii, is it not?
how does the data look on this address on a wii?
also, if it works without patching IOS, it be better to do this check before doing the ahbprot patching, not after it when doesn't matter any more.

Reply:
That address isn't out of bounds on console. It returns 0 on a Wii, or 0xCAFE on a vWii (when the app has the proper rights to access it; otherwise it returns 0 on vWii as well). On Dolphin, it returns 0xffff, making for an easy way to detect Dolphin. 
I tested this PR both on Dolphin and on Wii, and it did work on both.

Reply:
And I did the check afterwards because I only wanted to do that check if the HW_AHBPROT setup failed.

Reply:
on wiiu that address is the cpu's revision number registry, on the wii its probably undefined -> i can't find anything about it on wiibrew and dolphin assumes its the gpu's MMIO (which is probably incorrect as it doesn't go that high )

im not saying it doesn't work. im saying the check iffy in its nature.

as for the check, its better moved up. even if dolphin would suddenly give access to that memory range to access HW_AHBPROT (which it shouldn't, unless a title's TMD says it should) , you have nand access anyway afaik. if not, we can change it later.

Reply:
Well that code is used in a bunch of other homebrew as well to detect Wii vs. vWii (for example, [Nintendont](https://github.com/FIX94/Nintendont/blob/master/loader/include/global.h#L149), so if this would cause freezes or be unreliable on Wii, that would already be known by now. 

Okay, I'll move the check up to before the IOS patches are triggered. Assuming none of the IOS patches are needed on Dolphin.

Reply:
i had to look for a second, but why aren't you trying to open /dev/dolphin in IOS? 
you PR'd it into dolphin, and it should be , imo, a better check.
is it because that version isn't released yet?

Reply:
That version is released, but there's still a huge amount of people who don't update their Dolphin and are still on 5.0-stable (more than 3 years old) or older dev builds. /dev/dolphin was mainly for controlling Dolphin and finding out the exact Dolphin version, which we needed for the Wiimmfi patch. For the Priiloader, the exact version of Dolphin doesn't matter, just that it is Dolphin, so there is no reason to use the new way that doesn't work on all versions.

Reply:
ye ok, released as a build but not a version release iirc.
who , besides dev's , would benefit from this change anyway? do users actually use the installer in dolphin?
if it's only dev's then i'd prefer to have it open /dev/dolphin to check, since it doesn't effect end users and it's a much more clear check.

few more notes btw : 
- remove the printf. at that stage of the installer it's init'ing the system. that printf will confuse the console and it will look like 'initing system < newline > dolphin detected < newline> [ the rest of the text '. the initing system will still be there after init, and the dolphin detected doesn't have any added value for the end user. i'd keep the gprintf though. 
- if in dolphin its not going to do anything in that if, then its better to have the check for dolphin above the ahbprot check ( read32(0x0d800064) == 0xFFFFFFFF) ). basically if(dolphin) -> gprintf else if (read32(0x0d800064) == 0xFFFFFFFF) etc etc. much cleaner

Reply:
I don't know if users would actually install Priiloader in Dolphin. As of now, they wouldn't be able to do so, because the installer runs into an error. 
I'll remove the printf and move the check to before the HW_AHBPROT check. However to check for /dev/dolphin, I'd need to call ISFS_Initialize, and I don't know if that works properly at that location, given that there is lots of reloading / IOS stuff going on afterwards.

Reply:
why would you need to init ISFS? 
afaik the /dev/dolphin is a device, not a file. you can just open the device with IOS, so IOS_Open("/dev/dolphin",IPC_OPEN_NONE); works fine ( + IOS_Close if it opened correctly) . i didn't test this on a wii mind you, but in dolphin it works fine...

Reply:
Oh ... right. If that does work, it is indeed a cleaner solution. I'll try adding all that.

Reply:
Code is now checking for /dev/dolphin.

Reply:
ye, it libogc will try to boot SM if the HBC loader isn't there and iirc dolphin dislikes that sometimes. including when priiloader boots it XD

ill check again when im home and merge